api/tests/routes/backoffice/finance_test.py

```diff
@@ -894,3 +896,211 @@ def test_generate_invoices_when_task_already_running(self, authenticated_client)
+        finance_incident = finance_models.FinanceIncident.query.get(original_finance_incident.id)
+        assert finance_incident.forceDebitNote == original_finance_incident.forceDebitNote
+
+
+class CancelDebitNoteTest(PostEndpointHelper):
+    endpoint = "backoffice_web.finance_incidents.cancel_debit_note"
+    endpoint_kwargs = {"finance_incident_id": 1}
+    needed_permission = perm_models.Permissions.MANAGE_INCIDENTS
+
+    def test_cancel_debit_note(self, authenticated_client):
+        reimbursement_point = offerers_factories.VenueFactory(bookingEmail="pro@example.com")
+        venue = offerers_factories.VenueFactory(pricing_point="self", reimbursement_point=reimbursement_point)
+        booking = bookings_factories.ReimbursedBookingFactory(venue=venue, stock__offer__venue=venue)
+        original_event = finance_factories.UsedBookingFinanceEventFactory(booking=booking)
+        original_pricing = api.price_event(original_event)
+        original_pricing.status = finance_models.PricingStatus.INVOICED
+
+        now = datetime.datetime.utcnow()
+        finance_incident = finance_factories.IndividualBookingFinanceIncidentFactory(
+            booking=booking,
+            incident__status=finance_models.IncidentStatus.VALIDATED,
+            incident__forceDebitNote=True,
+            incident__venue=venue,
+        ).incident
+        api._create_finance_events_from_incident(
+            finance_incident.booking_finance_incidents[0], incident_validation_date=now
+        )
+
+        response = self.post_to_endpoint(
+            authenticated_client,
+            finance_incident_id=finance_incident.id,
+        )
+
+        assert response.status_code == 200
+        assert (
+            "Vous avez fait le choix de récupérer l'argent sur les prochaines réservations de l'acteur."
+            in html_parser.content_as_text(response.data)
+        )
+
+        assert len(mails_testing.outbox) == 1
+        assert (
+            mails_testing.outbox[0].sent_data["template"]
+            == TransactionalEmail.RETRIEVE_INCIDENT_AMOUNT_ON_INDIVIDUAL_BOOKINGS.value.__dict__
+        )
+        assert mails_testing.outbox[0].sent_data["To"] == "pro@example.com"
+        assert mails_testing.outbox[0].sent_data["params"] == {
+            "VENUE_NAME": venue.publicName,
+            "OFFER_NAME": booking.stock.offer.name,
+        }
+        updated_finance_incident = finance_models.FinanceIncident.query.get(finance_incident.id)
+        assert updated_finance_incident.forceDebitNote == False
+
+    @pytest.mark.parametrize(
+        "incident_status",
+        [
+            finance_models.IncidentStatus.CANCELLED,
+            finance_models.IncidentStatus.CREATED,
+        ],
+    )
+    def test_not_cancel_debit_note(self, authenticated_client, incident_status):
+        original_finance_incident = finance_factories.FinanceIncidentFactory(status=incident_status)
```

Comment:
Pour que la vérification de ce test soit juste, mieux vaut partir d'une situation de départ avec `forceDebitNote` à `True`, et vérifier à la fin : `assert finance_incident.forceDebitNote is True`.
Pareil pour l'autre test de cet _endpoint_.